### ISSUE TITLE
@Delegate annotation fails in plain Groovy src when referencing a Domain Object

### ISSUE THREAD

**Issue by timbonicus** [grails/grails-data-mapping#856]
Converting a Grails 2 project to Grails 3, we ran into an issue with some wrapper classes that use the @Delegate Groovy annotation:
```
Can't have an abstract method in a non-abstract class. The class 'test.delegate.annotation.TestWrapper' must be declared abstract or the method 'boolean org_grails_datastore_gorm_GormValidateable__skipValidate$get()' must be implemented.
 @ line 3, column 1.
```
I worked around it by having the wrapper implement GormValidateable, DirtyCheckable, and Validateable. I'm not sure if this is the best solution. I found [GROOVY-7288](https://issues.apache.org/jira/browse/GROOVY-7288) that looks like the same issue. Maybe it's a Groovy bug?

Here's a test project that reproduces the issue:
[test-delegate-annotation.zip](https://github.com/grails/grails-data-mapping/files/725334/test-delegate-annotation.zip)

**Comment by graemerocher** [comment]
Yeah looks the same. So we need https://issues.apache.org/jira/browse/GROOVY-7288 to be resolved first

**Comment by chrisbrookes** [comment]
Is this also expected to not work with using `@Delegate` against a POJO (i.e. something that is not  Grails domain class)? Doesn't seem to bind the incoming params, but there's no error either.

```
class MyCmd implements Validateable
{
    @Delegate
    Pojo pojo = new Pojo()

    static constraints = {
        // some constraints against Pojo
    }
}
```
If I add copy a field from Pojo into MyCmd, binding works as expected.
